### DISCUSSION TITLE
Suggestions on improving 0 digit detection

### DISCUSSION THREAD

**Discussion by gedger** [jomjol/AI-on-the-edge-device#3023] [Q&A]
My water meter is being read very successfully except for when the digit is showing a zero when it is frequently shown as NaN. I believe I have optimised the image and alignment although I still have a slight reflection on digit before the comma that I'm working on. Any suggestions for improving the detection of zeros?

![IMG_0205](https://github.com/jomjol/AI-on-the-edge-device/assets/8577703/3d5ccb25-1677-4498-bb2b-3bfaf52da468)
![IMG_0206](https://github.com/jomjol/AI-on-the-edge-device/assets/8577703/03ed2516-65f1-420d-9a66-449a11c06cbd)

**Comment by SybexX** [comment]
I have the same problem with the 0^^
The only thing that helped me to partially compensate for this problem was:
- Improve image quality (brightness, contrast...)
- zoom in to increase the size of the ROIs
Maybe changing the model could also help you

**Reply by gedger** [reply]
Thanks, have just tried zooming without any improvement. I'll have a think about improving the lighting although the digits look very clear to me, perhaps I should also investigate how to add my digits to the training set.

![IMG_0207](https://github.com/jomjol/AI-on-the-edge-device/assets/8577703/c7e379d2-d5dd-4498-9d75-18ace0f84105)

**Comment by SybexX** [comment]
I tried training, but it didn't really bring any improvement for me^^
A large difference between the number and the background is very important for recognizing, which unfortunately is not the case for you.
try this version, with it you can zoom a little better: https://github.com/jomjol/AI-on-the-edge-device/discussions/2926

**Comment by gedger** [comment]
I haven't been able to test out your improved zoom variant but I have found that using the dig-class100-0168_s2_q.tflite has improved matters over the digital-cont*** models that I had been using previously. I will test this for a few weeks and see how it performs.

Thanks for the suggestions.
